### ISSUE TITLE
private_outlet rule should ignore delegates properties

### ISSUE THREAD

**Issue by iDevid** [realm/SwiftLint#2373]
I love to use either storyboards to choose an element's delegate so i use the @IBOutlet attribute like this example:
```swift
@IBOutlet public weak var delegate: MyDelegate?
```
But a delegate should be also public to set it up by code, and this will break the private_outlet rule.
This should be valid also for dataSource properties.

What do you think?

**Comment by marcelofabri** [comment]
This probably should be a configuration.

**Comment by damien-nd** [comment]
I'm facing the same issue. Is this plan for future release? Or do you have some files or tutorials to point me in right direction?

**Comment by marcelofabri** [comment]
> I'm facing the same issue. Is this plan for future release? Or do you have some files or tutorials to point me in right direction?

You can take a look at `PrivateOutletRule.swift`, `PrivateOutletRuleConfiguration.swift` and `PrivateOutletRuleTests.swift`.
